code/clustering.py

```
@@ -0,0 +1,31 @@
+import pandas as pd
+from sklearn.cluster import KMeans
+from sklearn.decomposition import PCA
+import matplotlib.pyplot as plt
+
+# Load embedded data
+df = pd.read_pickle("job_embeddings.pkl")
+
+# Combine embeddings into a single matrix
+embedding_cols = [col for col in df.columns if "embedding" in col]
+X = np.hstack(df[embedding_cols].values)
```

Ruff (0.8.2)

Undefined name `np`

(F821)